src/Notifynder/Parsers/NotificationParser.php

```diff
@@ -20,14 +20,19 @@ class NotificationParser
     /**
      * Parse a notification and return the body text.
      *
-     * @param array|ModelNotification|BuilderNotification $notification
+     * @param ModelNotification $notification
      * @param int $categoryId
      * @return string
      * @throws ExtraParamsException
      */
     public function parse($notification, $categoryId)
     {
-        $category = NotificationCategory::findOrFail($categoryId);
+        $category = $notification->category;
+        if (is_null($category)) {
+            throw (new ModelNotFoundException)->setModel(
```

Comment:
I agree we should remove the categoryId parameter.
https://github.com/fenos/Notifynder/issues/280#issuecomment-310856167